### PR TITLE
Use mojom namespace for ledger structs

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/25292

Notes:

* I left the `mojom_structs.h` header for now, since it combines definitions from two mojo files, and I didn't want to update includes any more than necessary.
* Aliases for collections (mostly `vector`) of mojo types have been removed.
* This is basically a search-and-replace textual change, along with subsequent lint fixes.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [x] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [x] New files have MPL-2.0 license header
- [x] Adequate test coverage exists to prevent regressions
- [x] Major classes, functions and non-trivial code blocks are well-commented
- [x] Changes in component dependencies are properly reflected in `gn`
- [x] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [x] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

Covered by existing tests.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution.h [hunk 1]
```diff
@@ -90,80 +88,68 @@ class Contribution {
...
+  void ContributionCompletedSaved(const mojom::Result result,
+                                  const std::string& contribution_id);
 
-  void OnProcessContributionQueue(type::ContributionQueuePtr info);
+  void OnProcessContributionQueue(mojom::ContributionQueuePtr info);
 
   void CheckNotCompletedContributions();
 
-  void NotCompletedContributions(type::ContributionInfoList list);
+  void NotCompletedContributions(std::vector<mojom::ContributionInfoPtr> list);
```
[the remark this answers is not in the document]
Comment (zenparsing) [reply vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution.h:103]: Good question. Prior to this change `type::ContributionInfoList` was just an alias for `std::vector<mojom::ContributionInfoPtr>`, and I didn't want to change any underlying signatures - I only wanted to rename things. If there are signatures that need to be changed, we should probably create an issue and do that separately.

[the remark this answers is not in the document]
Comment (emerick) [reply vendor/bat-native-ledger/src/bat/ledger/internal/contribution/contribution.h:103]: Yeah, that makes sense to me.

**Review by emerick — APPROVED** [review]
LGTM, I mostly spot-checked several files in all of the main Rewards components.

**Review by kylehickinson — APPROVED** [review]
iOS++

**Review by szilardszaloki — APPROVED** [review]
Did the same as @emerick. I can't really see any pitfalls with textually replacing these namespaces. A failure here would result in a compile-time error.

**Review by aseren — APPROVED** [review]
ads ++

**Pull request merged**